frontstage/controllers/case_controller.py

```diff
@@ -261,3 +212,35 @@ def validate_case_category(category):
         raise InvalidCaseCategory(category)
 
     logger.debug('Successfully validated case category', category=category)
+
+
+def get_cases_for_list_type_by_party_id(party_id, list_type='todo'):
+    logger.debug('Get cases for party for list', party_id=party_id, list_type=list_type)
+
+    business_cases = get_cases_by_party_id(party_id)
+    cases = filter_cases_by_case_group(business_cases)
+    history_statuses = ['COMPLETE', 'COMPLETEDBYPHONE', 'NOLONGERREQUIRED']
+    if list_type == 'history':
+        filtered_cases = [business_case
+                          for business_case in cases
+                          if business_case['caseGroup']['caseGroupStatus'] in history_statuses]
+    else:
+        filtered_cases = [business_case
+                          for business_case in cases
+                          if business_case['caseGroup']['caseGroupStatus'] not in history_statuses]
+
+    logger.debug("Successfully retrieved cases for party survey list", party_id=party_id, list_type=list_type)
+    return filtered_cases
+
+
+def filter_cases_by_case_group(cases):
+    logger.debug("Attempting to remove multiple cases for one case group")
+    for key, group in itertools.groupby(cases, key=lambda x: x['caseGroup']['id']):
+        group = list(group)
+        if len(group) > 1:
+            sorted_group = sorted(group, key=lambda k: k['createdDateTime'], reverse=True)
+            for i in sorted_group:
+                if sorted_group.index(i) != 0:
+                    cases.remove(i)
```

Comment:
This can be simplified using generators I think
```
grouped_cases = (list(group)
                 for _, group in itertools.groupby(cases, key=lambda x: x['caseGroup']['id']))
return (sorted(group, key=lambda k: k['createdDateTime'], reverse=True)[0]
        for group in grouped_cases)
```